homeassistant/components/rejseplanen/sensor.py

```diff
@@ -6,252 +6,570 @@
+        }
+
+        departures_data.append(departure_info)
+
+    return {
+        "departures": departures_data,
+        "total_departures": len(current_departures),
+        "next_departure_in": departures_data[0]["due_in"] if departures_data else None,
+        "last_updated": datetime.now(tz).isoformat(),
+    }
 
-    The timestamp should be in the format day.month.year hour:minute
-    """
-    diff = datetime.strptime(timestamp, "%d.%m.%y %H:%M") - dt_util.now().replace(
-        tzinfo=None
-    )
 
-    return int(diff.total_seconds() // 60)
+def _get_empty_departure_attributes() -> dict[str, Any]:
+    """Return empty attributes structure when no departure data available."""
+    return {
+        ATTR_STOP_ID: None,
+        ATTR_STOP_NAME: None,
+        ATTR_FINAL_STOP: None,
+        ATTR_TRACK: None,
+        ATTR_PLANNED_TIME: None,
+        ATTR_REALTIME_TIME: None,
+        ATTR_DUE_IN: None,
+        ATTR_DUE_AT: None,
+        ATTR_SCHEDULED_AT: None,
+        ATTR_REAL_TIME_AT: None,
+        ATTR_IS_CANCELLED: None,
+        ATTR_DELAY_MINUTES: None,
+        "has_realtime": None,
+        "line_type": None,
+        "operator": None,
+    }
+
+
+# SENSORS tuple definition
+SENSORS: tuple[RejseplanenSensorEntityDescription, ...] = (
+    RejseplanenSensorEntityDescription(
+        key="next_departure",
+        translation_key="next_departure",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda departures: (
+            _get_departure_timestamp(departures, 0) if departures else None
+        ),
+        attr_fn=lambda departures: (
+            _get_departure_attributes(departures, 0)
+            if departures
+            else _get_empty_departure_attributes()
+        ),
+    ),
+    RejseplanenSensorEntityDescription(
+        key="departures",
+        translation_key="departures",
+        value_fn=lambda departures: len(_get_current_departures(departures)),
+        attr_fn=_get_departures_list_attributes,
+    ),
+    RejseplanenSensorEntityDescription(
+        key="next_line",
+        translation_key="next_line",
+        value_fn=lambda departures: (
+            _get_current_departures(departures)[0].name
+            if _get_current_departures(departures)
+            else None
+        ),
+    ),
+    RejseplanenSensorEntityDescription(
+        key="next_direction",
+        translation_key="next_direction",
+        value_fn=lambda departures: (
+            _get_current_departures(departures)[0].direction
+            if _get_current_departures(departures)
+            else None
+        ),
+    ),
+    RejseplanenSensorEntityDescription(
+        key="next_track",
+        translation_key="next_track",
+        value_fn=lambda departures: (
+            _get_current_departures(departures)[0].rtTrack
+            or _get_current_departures(departures)[0].track
+            if _get_current_departures(departures)
+            else None
+        ),
+    ),
+)
```

Comment:
Regarding entities and attributes

Ideally we have little to no attributes. A lot of these attributes are subject to automations and attributes simply do not provide a nice way to automate. Furthermore, they are all stored in the database as well, so we want to minimze them by exploding them into more entities.